### PR TITLE
To_Do_List Code review

### PR DESCRIPTION
**Pull request by ntzwilly**
#  Code Review exercise

### CODE REVIEW

File: src/crud.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  for (let i = 0; i < todoTasks.length; i++) {
+    todoTasks[i].id = i;
+    todoTasks[i].index = i + 1;
+  }
+
+  localStorage.clear();
+  localStorage.setItem('ToDo', JSON.stringify(todoTasks));
+}
+
+export function clearTasks() {
```
Comment (ahmetbozaci) [inline src/crud.js:55]: - [ ] You can `export` all functions end of `crud.js file` in one line. Also you can refactors functions to arrow functions. You can check [this](https://stackoverflow.com/questions/22939130/when-should-i-use-arrow-functions-in-ecmascript-6)

File: src/crud.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  form.addEventListener('submit', () => {
+    const newToDo = taskInput.value;
+    if (newToDo) {
+      todoTasks.push({
+        description: newToDo,
+        completed: false,
+        index: todoTasks.length + 1,
+        checked: false,
+        id,
+      });
```
Comment (ahmetbozaci) [inline src/crud.js:22]: You can create new object for that. And you can insert this `object`  into` todoTasks.`
```
const newTodoLıst = { description: newTodo, .... )
todoTasks.push(newTodoLıst)
```

**Review by ahmetbozaci — CHANGES_REQUESTED** [review]
# STATUS: Changes Required  ♻️ 
## Hello @ntzwilly ✋🏼  Your project is great
### ♻️  However there are some issue you should consider :♻️ 
_**Check the comment about that  under the review.**_

### Cheers and Happy coding! 💻 👏
cat on laptop


___
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_
___
_If anything seems to be unclear to you, reach out and ask anything by commenting on this Pull Request and mention me as [ahmetbozaci](https://github.com/ahmetbozaci), I will receive a notification and will respond._

File: src/style.css [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+main {
+  margin: 200px auto;
+  width: 80%;
+  box-shadow:
+    5px 0 15px -4px rgba(0, 0, 0, 0.8),
+    -5px 0 8px -4px rgba(0, 0, 0, 0.8);
+}
+
+.todo-container {
+  display: flex;
```
[the remark this answers is not in the document]
Comment (ntzwilly) [reply src/style.css:22]: Thank you, I made the requested change

File: src/crud.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    }
+  });
+}
+
+export function editTask(input, elem, form) {
+  let newValue = '';
+  input.setAttribute('name', elem.id);
+  input.addEventListener('input', (e) => {
+    newValue = e.target.value;
+    e.preventDefault();
```
[the remark this answers is not in the document]
Comment (ntzwilly) [reply src/crud.js:33]: Thank you, changed e to (event) to be more descriptive

File: src/crud.js [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  });
+
+  form.addEventListener('submit', (e) => {
+    const p = e.target[0].name;
+
+    todoTasks[p].description = newValue;
+    savedList();
+  });
+}
```
[the remark this answers is not in the document]
Comment (ntzwilly) [reply src/crud.js:42]: I changed e to element :smiley:

File: src/crud.js [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+  btnClear.addEventListener('click', () => {
+    const pendingTasks = todoTasks.filter((item) => item.checked !== true);
+    for (let i = 0; i < pendingTasks.length; i++) {
+      pendingTasks[i].id = i;
+      pendingTasks[i].index = i + 1;
+    }
+    localStorage.setItem('ToDo', JSON.stringify(pendingTasks));
+    window.location.reload();
+  });
+}
```
[the remark this answers is not in the document]
Comment (ntzwilly) [reply src/crud.js:65]: Thank you, I now use forEach. No need to use Array.length anymore :smile:

File: src/crud.js [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+  todoTasks.splice(elt, 1);
+  for (let i = 0; i < todoTasks.length; i++) {
+    todoTasks[i].id = i;
+    todoTasks[i].index = i + 1;
+  }
+
+  localStorage.clear();
+  localStorage.setItem('ToDo', JSON.stringify(todoTasks));
+}
+
```
[the remark this answers is not in the document]
Comment (ntzwilly) [reply src/crud.js:54]: I renamed elt to element and I use now forEach loop

File: src/crud.js [hunk 8]
```diff
@@ -0,0 +1,65 @@
+/* eslint-disable import/no-cycle */
+/* eslint-disable no-plusplus */
+import { savedList } from './interactive.js';
+import {
+  todoTasks, taskInput, btnClear,
+} from './index.js';
+
+export function createTask() {
+  const form = document.querySelector('.to-do');
```
[the remark this answers is not in the document]
Comment (ntzwilly) [reply src/crud.js:9]: Thank you, I have renamed the form. It is now todoForm

File: src/crud.js [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+
+  const id = todoTasks.length;
```
[the remark this answers is not in the document]
Comment (ntzwilly) [reply src/crud.js:11]: Thank you for your comment. The id is the id of the element.

File: src/crud.js (hunk 1 above)
Comment (ntzwilly) [reply src/crud.js:55]: The functions are now exported in a single line

File: src/crud.js (hunk 2 above)
Comment (ntzwilly) [reply src/crud.js:22]: Thank you, I have made this change

**Pull request closed without merging**